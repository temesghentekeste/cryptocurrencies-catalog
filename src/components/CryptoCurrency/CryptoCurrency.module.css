.cryptoCurrencyCard {
  width: 15rem;
  margin-bottom: 1.4rem;
  margin-right: 2rem;
  padding: 1rem;
  border: 1px solid gray;
  background: #444;
  color: white;
  border-radius: 5px;
}

.cryptoCurrencyCard__symbol {
  margin-bottom: 1rem;
  padding: 0.4rem 0.6rem;
  background: #998695;
  color: #fff;
  font-weight: 700;
}

.cryptoCurrencyCard__body {
  display: flex;
  justify-content: space-between;
  flex-wrap: wrap;
}

.cryptoCurrencyCard button {
  width: 100%;
  padding: 0.6rem 0.8rem;
  margin: 1rem 0 0;
  background: rgb(75, 95, 73);
  border: none;
  border-radius: 5px;
  font-weight: 700;
  color: #fff;
}

.cryptoCurrencyCard img {
  display: block;
  width: 90%;
  margin: 1rem auto;
}

@media screen and (max-width: 982px) {
  .cryptoCurrencyCard {
    width: 20rem;
  }
}

@media screen and (max-width: 850px) {
  .cryptoCurrencyCard {
    width: 18rem;
  }
}

@media screen and (max-width: 768px) {
  .cryptoCurrencyCard {
    width: 100%;
  }
}
