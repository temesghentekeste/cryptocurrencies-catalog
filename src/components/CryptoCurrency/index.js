import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import styles from './CryptoCurrency.module.css';
import mainStyle from '../../index.module.css';

const CryptoCurrency = ({
  id, symbol, name, price, image,
}) => (
  <Link to={`/cryptoquote/${id}`}>
    <div className={styles.cryptoCurrencyCard} data-testid="cryptocurrency">
      <header data-testid="cryptocurrency-header">
        <p className={mainStyle.badge} data-testid="cryptocurrency-name">
          {name}
        </p>
        <p
          className={styles.cryptoCurrencyCard__symbol}
          data-testid="cryptocurrency-symbol"
        >
          {symbol}
        </p>
      </header>

      <div className={styles.cryptoCurrencyCard__body}>
        <span data-testid="cryptocurrency-price">
          Price: $
          {price}
        </span>
        <img src={image} alt={name} />
      </div>

      <footer>
        <button type="button">More...</button>
      </footer>
    </div>
  </Link>
);

CryptoCurrency.propTypes = {
  symbol: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  price: PropTypes.number.isRequired,
  image: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
};
export default CryptoCurrency;
