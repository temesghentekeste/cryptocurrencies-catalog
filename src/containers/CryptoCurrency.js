import React from 'react';

const CryptoCurrency = ({
  symbol,
  name,
  currency,
  stockExchange,
  exchangeShortName,
}) => (
  <div className="cryptoCurrency">
    <header>
      <p>{name}</p>
      <p className="cryptoCurrency__symbol">{symbol}</p>
    </header>

    <footer>
      <span>{currency}</span>
      <span>{stockExchange}</span>
      <span>{exchangeShortName}</span>
    </footer>
  </div>
);

export default CryptoCurrency;
