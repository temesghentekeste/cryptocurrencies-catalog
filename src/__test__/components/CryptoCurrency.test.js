import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import CryptoCurrency from '../../components/CryptoCurrency';

let getByTestId;

beforeEach(() => {
  const component = render(
    <Router>
      <CryptoCurrency
        id="Bitcoin"
        symbol="btc"
        name="Bitcoin"
        price={37255}
        image="https://ichef.bbci.co.uk/news/976/cpsprodpb/6127/production/_117317842_065760657.jpg"
      />
    </Router>,
  );
  getByTestId = component.getByTestId;
});

it('renders the CryptoCurrency component', () => {
  const component = renderer
    .create(
      <Router>
        <CryptoCurrency
          id="Bitcoin"
          symbol="btc"
          name="Bitcoin"
          price={37255}
          image="https://ichef.bbci.co.uk/news/976/cpsprodpb/6127/production/_117317842_065760657.jpg"
        />
      </Router>,
    )
    .toJSON();

  expect(component).toMatchSnapshot();
});

test('should render CryptoCurrency component with correct name', () => {
  const cryptoCurrencyEl = getByTestId('cryptocurrency-name');

  expect(cryptoCurrencyEl).toHaveTextContent('Bitcoin');
});

test('should render CryptoCurrency component with correct name', () => {
  const cryptoCurrencyEl = getByTestId('cryptocurrency-name');

  expect(cryptoCurrencyEl).not.toHaveTextContent('eth');
});

test('should render CryptoCurrency component with correct price', () => {
  const cryptoCurrencyEl = getByTestId('cryptocurrency-price');

  expect(cryptoCurrencyEl).toHaveTextContent('$37255');
});

test('should render CryptoCurrency component with correct symbol', () => {
  const cryptoCurrencyEl = getByTestId('cryptocurrency-symbol');

  expect(cryptoCurrencyEl).not.toHaveTextContent('0');
});

test('should render CryptoCurrency component with correct symbol', () => {
  const cryptoCurrencyEl = getByTestId('cryptocurrency-symbol');

  expect(cryptoCurrencyEl).toHaveTextContent('btc');
});

test('should render CryptoCurrency component with correct symbol', () => {
  const cryptoCurrencyEl = getByTestId('cryptocurrency-symbol');

  expect(cryptoCurrencyEl).not.toHaveTextContent('eth');
});
