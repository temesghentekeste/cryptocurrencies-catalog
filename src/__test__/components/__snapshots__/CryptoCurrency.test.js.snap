// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the CryptoCurrency component 1`] = `
<a
  href="/cryptoquote/Bitcoin"
  onClick={[Function]}
>
  <div
    className="cryptoCurrencyCard"
    data-testid="cryptocurrency"
  >
    <header
      data-testid="cryptocurrency-header"
    >
      <p
        className="badge"
        data-testid="cryptocurrency-name"
      >
        Bitcoin
      </p>
      <p
        className="cryptoCurrencyCard__symbol"
        data-testid="cryptocurrency-symbol"
      >
        btc
      </p>
    </header>
    <div
      className="cryptoCurrencyCard__body"
    >
      <span
        data-testid="cryptocurrency-price"
      >
        Price: $
        37255
      </span>
      <img
        alt="Bitcoin"
        src="https://ichef.bbci.co.uk/news/976/cpsprodpb/6127/production/_117317842_065760657.jpg"
      />
    </div>
    <footer>
      <button
        type="button"
      >
        More...
      </button>
    </footer>
  </div>
</a>
`;
